src/AutoRest.CSharp/Common/Input/CadlInputEnumTypeConverter.cs

```diff
@@ -103,7 +121,17 @@ private static IReadOnlyList<InputEnumTypeValue> normalizeValues(IReadOnlyList<I
                 case InputTypeKind.Float32:
                     foreach (var value in allowedValues)
                     {
-                        concreteValues.Add(new InputEnumTypeFloatValue(value.Name, (float)value.Value, value.Description));
+                        switch (value.Value)
+                        {
+                            case Int32 i:
+                                concreteValues.Add(new InputEnumTypeFloatValue(value.Name, (float)(Int32)i, value.Description));
```

Comment:
We could implicitly convert `int` to `float` therefore we do not need the cast here
Please ignore this if I am wrong
```suggestion
                                concreteValues.Add(new InputEnumTypeFloatValue(value.Name, i, value.Description));
```